### ISSUE TITLE
Regression: "Each" for start value of array not recognized anymore

### ISSUE THREAD

**Issue by bilderbuchi** [OpenModelica/OpenModelica#12838]
### Description
I am setting the start values of an array in this model:
```modelica
model StartEach
Real[3] out;
parameter Real[3] someval(each start=42.0);
equation
out = 1.0 .- someval;

end StartEach;
```
This works as expected in 1.23.1, and produces `out` values of -41.

In 1.24.0 nightlies, currently dev-250-gb4971595da, this does not work anymore, as I get a translation error
> [Non-array modification '42.0' for array component 'someval', possibly due to missing 'each'.]

Obviously, the `each` is not missing in the source code. 

Note: This also happens when uncommenting the declaration and equation involving `out`, which could further reduce the MWE size (but then we get another warning.

### Expected
I expect this model to translate correctly, and probably warn that we are using the available start value for `someval`.

**Comment by bilderbuchi** [comment]
mybe related: #7777 #12359

**Comment by perost** [comment]
Fixed by #12839.

The issue is that you can't use `each` to set the values of an array like that (`Real[3] someval(each ??? = 42.0)`), so just copying the binding from the start attribute to the component doesn't work.

It might be possible to handle it anyway when typing the bindings, but that's already complicated enough without extra special cases. Instead I just changed it to expand the binding into e.g. `fill(42.0, size(someval, 1))` in this case before applying it to the component.

**Comment by bilderbuchi** [comment]
Thank you, I can confirm that it works now.

**Comment by bilderbuchi** [comment]
@perost I uncovered a possible shortcoming of this fix, please see #12842 and comment if you can. :-)
